src/slider/slider.js

```diff
@@ -0,0 +1,225 @@
+
+  onMouseUp = (e: MouseEvent) => {
+    if (this.state.isThumbMoving) {
+      this.setState({
+        isThumbMoving: false,
+        currentThumb: -1,
+        currentMove: 0,
+      });
+    }
+  };
+
+  onMouseMove = (e: MouseEvent) => {
+    if (this.state.isThumbMoving) {
+      this.onMove(e.movementX, e);
+    }
+  };
+
+  addDocumentMouseEvents() {
+    if (__BROWSER__) {
+      this.onMouseMoveListener = document.addEventListener(
+        'mousemove',
+        this.onMouseMove,
+      );
+      this.onMouseUpListener = document.addEventListener(
+        'mouseup',
+        this.onMouseUp,
+      );
+    }
+  }
+
+  removeDocumentEvents() {
+    this.onMouseMoveListener && this.onMouseMoveListener.remove();
+    this.onMouseUpListener && this.onMouseUpListener.remove();
+  }
+
+  render() {
+    const {overrides = {}, range, value} = this.props;
+    const {thumbRefs} = this.state;
+    const [Root, rootProps] = getOverrides(overrides.Root, StyledRoot);
+    const [Axis, axisProps] = getOverrides(overrides.Axis, StyledAxis);
+    const [AxisRange, axisRangeProps] = getOverrides(
+      overrides.AxisRange,
+      StyledAxisRange,
+    );
+    const [Thumb, thumbProps] = getOverrides(overrides.Thumb, StyledThumb);
+    const [Tick, tickProps] = getOverrides(overrides.Tick, StyledTick);
+    const [TickBar, tickBarProps] = getOverrides(
+      overrides.TickBar,
+      StyledTickBar,
+    );
+    const sharedProps = this.getSharedProps();
+    return (
+      <Root {...sharedProps} onClick={e => this.onAxisClick(e)} {...rootProps}>
+        <Axis {...sharedProps} {...axisProps}>
+          {value.map((thumb, $index) => (
+            <React.Fragment key={$index}>
+              <AxisRange $index={$index} {...sharedProps} {...axisRangeProps} />
+              <Thumb
+                $ref={thumbRefs[$index]}
+                key={$index}
+                $index={$index}
+                {...thumbProps}
+                {...sharedProps}
+                onMouseDown={e => this.onMouseDown(e, $index)}
+              />
+            </React.Fragment>
+          ))}
+        </Axis>
+        <TickBar {...sharedProps} {...tickBarProps}>
+          {range.map((tick, $index) => (
+            <Tick key={$index} $index={$index} {...sharedProps} {...tickProps}>
+              {tick}
+            </Tick>
+          ))}
+        </TickBar>
+      </Root>
+    );
+  }
+
+  onMove(movementX: number, e: SyntheticEvent<HTMLElement> | MouseEvent) {
+    const {currentThumb, currentMove} = this.state;
+    const move = currentMove + movementX;
+    const {$value, $max, $min, $isRange} = this.getSharedProps();
+    const value = $value.slice();
+    let {onChange, step} = this.props;
+    const axisSizeInPixels = parseInt(
+      window.getComputedStyle(this.domNode).width,
+    );
+    const axisSize = $max - $min;
+    step = step ? Slider.scale(step, axisSize, axisSizeInPixels) : step;
+    const scaledStep = step ? (move >= 0 ? step : -step) : move;
```

Comment:
maybe this is just me, but can we break it down into a more easily understandable format?

Reply:
changed to be more readable